Python-packages/covidcast-py/covidcast/plotting.py

```diff
@@ -210,8 +239,31 @@ def _join_county_geo_df(data: pd.DataFrame,
                               right_on="state", sort=True)
         # if no county value present, us the megacounty values
         for c in input_cols:
-            merged[c] = [j if pd.isna(i) else i for i, j in zip(merged[f"{c}_x"], merged[f"{c}_y"])]
+            merged[c] = merged[f"{c}_x"].combine_first(merged[f"{c}_y"])
     # use the full county FIPS list in the return
-    merged[county_col] = [j if pd.isna(i) else i for i, j in zip(merged.GEOID, merged[county_col])]
+    merged[county_col] = merged.GEOID.combine_first(merged[county_col])
     merged.rename(columns={"STATEFP": "state_fips"}, inplace=True)
     return gpd.GeoDataFrame(merged[input_cols + ["state_fips", "geometry"]])
+
+
+def _join_msa_geo_df(data: pd.DataFrame,
+                     msa_col: str,
+                     geo_info: gpd.GeoDataFrame,
+                     join_type: str = "right") -> gpd.GeoDataFrame:
+    """Join DF information to polygon information in a GeoDF at the state level.
+
+    For MSAs which span multiple states, the first state in the name is returned for the state FIPS.
+
+    :param data: DF with state info
+    :param msa_col: cname of column in `data` containing state info to join on
+    :param geo_info: GeoDF of state shape info read from Census shapefiles
+    :return: ``data`` with cbsa polygon and state fips joined.
+    """
+    geo_info = geo_info[geo_info.LSAD == "M1"]
+    input_cols = list(data.columns)
+    merged = data.merge(geo_info, how=join_type, left_on=msa_col, right_on="GEOID")
+    # use full state list in the return
+    merged[msa_col] = merged.GEOID.combine_first(merged[msa_col])
+    # get the first state, which will be the first two characters after the comma and whitespace
+    merged["state_fips"] = [STATE_ABBR_TO_FIPS.get(i.split(",")[1][1:3]) for i in merged.NAME]
```

Comment:
Not sure this is the exact behaviour we want, so open to discussion here. There are cases where the MSA name is ` New York-Newark-Jersey City, NY-NJ-PA `, in which case this takes NY to be the state. Theres 61 cases of one MSA mapping to multiple states. 

Also as a reminder, the main reason we use this is so that `_project_and_transform` can know where AK, HI, and PR are and transform and move accordingly. I can imagine an alternate implementation where we don't return state FIPS and instead let `_project_and_transform` take in a `geo_column` and `geo_type` argument (both of which `get_geo_df` knows), and applies a `geo_type` specific mapping in order to know what's in AK/HI/PR and what isn't. Now that I say that out loud, I kind of like that.....I think it's relatively low pri but I like the idea of cleaning up the interface for the `get_geo_df` function to only return polygons.